src/librustc_mir_build/hair/pattern/_match.rs

```diff
@@ -291,30 +291,39 @@ impl<'tcx> LiteralExpander<'tcx> {
                     }
                     Scalar::Raw { .. } => {
                         let layout = self.tcx.layout_of(self.param_env.and(rty)).unwrap();
-                        if layout.is_zst() {
-                            // Deref of a reference to a ZST is a nop.
-                            ConstValue::Scalar(Scalar::zst())
-                        } else {
-                            // FIXME(oli-obk): this is reachable for `const FOO: &&&u32 = &&&42;`
```

Comment:
Was this comment incorrect before?

Reply:
apparently, we have a test for this, but the ICE is a different one somewhere in the middle of pattern matching. I've so far been unable to figure it out.